src/couch/src/couch_server.erl

```diff
@@ -101,9 +101,14 @@ open(DbName, Options0) ->
     end.
 
 update_lru(DbName, Options) ->
-    case lists:member(sys_db, Options) of
-        false -> gen_server:cast(couch_server, {update_lru, DbName});
-        true -> ok
+    case config:get_boolean("couchdb", "update_lru_on_read", false) of
```

Comment:
This would add one extra `ets:lookup` here (which we probably can accept). A hacky alternative could be to extend `#entry` record so we can get the value of the parameter before sending message to couch_server.

Reply:
Yeah... I try to pretend that ETS lookup is fast enough to not be relevant. We do config lookups all over the place, so if we're concerned about that I think we need to deal with that as a bigger problem that working around individual lookups. For instance, one option would be to bootstrap new http request pids with a prebuilt config table that has all the options they need to lookup so no concurrent ETS interactions are necessary. Although we would need to do a fair bit of testing to actually determine whether config lookups are a bottleneck.

Reply:
We could do the mochiglobal pattern eventually. 

But last time I had worried about `ets:lookup` I had measured it at less than 5 microseconds.

Was curious and tried it again my mac laptop:

```
f(Tries), f(T), Tries = lists:seq(1,1000000), {T, ok} = timer:tc(fun() -> lists:foreach( fun(_) -> config:get_boolean("couchdb", "update_lru_on_read", false) end, Tries) end), T/1.0e6.
0.955564
```

On a linux server:
```
f(Tries), f(T), Tries = lists:seq(1,1000000), {T, ok} = timer:tc(fun() -> lists:foreach( fun(_) -> config:get_boolean("couchdb", "update_lru_on_read", false) end, Tries) end), T/1.0e6.
1.264781
```

Granted neither the laptop not the server were very busy, so maybe with a larger number of lock acquisitions it looks much worse.